packages/server/src/user/user.service.ts

```diff
@@ -86,12 +87,20 @@ export class UserService {
    * @returns `User` object, or throw `NotFoundError` when not exist
    */
   async findUserByUsername(projectId: string, username: string): Promise<User> {
-    return this.prisma.user.findFirstOrThrow({
```

Comment:
Changing the function to return `null` instead of throwing an error has impacts elsewhere in the code like in the `user.controller`.

Reply:
`findUserByUsername` and `findUserByEmail` aren't called in controller. I did, however, test `auth.service.validateUsername` and `auth.service.validateEmail` which call them.